image-processing/UI.py

```diff
@@ -1,484 +1,202 @@
-                            try:
-                                row = int(row_input)
-                                if row < 1:
-                                    print("Invalid input. Please ensure a positive integer is entered.")
-                                    continue
+            if int(row) > len(scanned): 
+                print(f'Row number index out of range. Please enter an integer within the bounds 0 to {len(scanned)}.\n')
+                self.input_row(scanned)
+            
+            return int(row)
+        
+        # elif row.lower() == "back":
+        #     self.input_sigma()
+        
+        else:
+            print("Invalid input. Please ensure a positive whole number is entered.\n")
+            self.input_row(scanned)
 
-                            except:
-                                print("Invalid input. Please ensure a whole number is entered.")
-                                continue
+    def quit_program(self):
+        end_program = input("Respond 'quit' if you have finished and would like to quit the program. \nOr, respond 'again' if you would like to run the program again: ").lower()
 
-                            original = scans_saved[scan][0]
-                            try:
-                                finder = linefinder(original, blur, row)
-                            
-                            except:
-                                print('Row index was out of bounds of the sample. The maximum row value is {}. Please try again'.format(len(original)))
-                                continue
-                            
-                            break
+        if end_program == 'quit':
+            sys.exit()
 
-                        while True:
-                            baseline_input = input('What severity out of 10 is required to fail a sample? \n Recommendations: 4 for Carbon Veil \n 6 for Metal-Coated Carbon Veil \n Enter Value:  ')
-                            try:
-                                baseline = float(baseline_input)    #got rid of divide by 2 for now
-                                if baseline < 1:
-                                    print("Invalid input. Please ensure a positive number is entered.")
-                                    continue
-                                break
+        elif end_program == 'again':
+            '''A new instance is created every time this function runs, overwriting the last instance. So no need to empty files[] list.'''
+            new_run = UserInterface()
+            new_run.main()
 
-                            except:
-                                print("Invalid input. Please ensure a number is entered.")
-                                continue
+        else:
+            print('Sorry, that response was not recognised, please try again.\n')
+            self.quit_program()
+            
 
+    def main(self):
+        no_of_files = len(glob.glob(f"{self.path}\*.jpeg"))
```

Comment:
just wanted to add that this is *very* cool

Reply:
creds to uhh not me woops